archive/tar_test.go

```diff
@@ -1,4 +1,4 @@
-// +build !windows
+// +build !windows,!darwin
```

Comment:
Do you know what didn't work on Darwin for this?

Reply:
```
--- FAIL: TestBreakouts (9.18s)
    --- FAIL: TestBreakouts/HardlinkSymlinkBeforeCreateTarget (8.78s)
        tar_test.go:934: Failed to apply tar: link /var/folders/nx/lmjd5hsx3r3gwyr3tgw14q140000gn/T/test-writer-to-tar-027924874/localpasswd /var/folders/nx/lmjd5hsx3r3gwyr3tgw14q140000gn/T/test-writer-to-tar-027924874/localpasswd-dup: operation not permitted
    --- FAIL: TestBreakouts/HardlinkSymlinkRelative (0.01s)
        tar_test.go:934: Failed to apply tar: link /var/folders/nx/lmjd5hsx3r3gwyr3tgw14q140000gn/T/test-writer-to-tar-068255073/passwdlink /var/folders/nx/lmjd5hsx3r3gwyr3tgw14q140000gn/T/test-writer-to-tar-068255073/localpasswd: no such file or directory
    --- FAIL: TestBreakouts/HardlinkSymlinkAbsolute (0.09s)
        tar_test.go:934: Failed to apply tar: link /var/folders/nx/lmjd5hsx3r3gwyr3tgw14q140000gn/T/test-writer-to-tar-814375756/passwdlink /var/folders/nx/lmjd5hsx3r3gwyr3tgw14q140000gn/T/test-writer-to-tar-814375756/localpasswd: operation not permitted
FAIL
FAIL	github.com/containerd/containerd/archive	13.724s
```

Reply:
I didn't dive in a lot what exactly is wrong here, for now I just want to enable tests in general.
We can follow up with this to run more tests.